sunpy/map/maputils.py

```diff
@@ -1,7 +1,252 @@
+    the center of the Sun.  If at least one pixel is on disk and at least one
+    pixel is off disk, the function returns True.  Otherwise, the function
+    returns False. Note that this function will also true if the entire solar
+    limb is within the field of view of the map.  Note also that in the case
+    of coronagraph images the limb itself need not be observed.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
+
+    Returns
+    -------
+    contains_limb : `~bool`
+        Returns True If at least one pixel is on disk and at least one pixel
+        is off disk.
+    """
+    on_disk = coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs)
+    return np.logical_and(np.any(on_disk), np.any(~on_disk))
+
+
+@u.quantity_input
+def coordinate_is_on_disk(coordinate, scale: u.arcsecond):
+    """
+    Checks if the helioprojective Cartesian coordinate is on disk.
+
+    The check is performed by comparing the coordinate's distance
+    from the center of the Sun to the solar radius.
+
+    Parameters
+    ----------
+    coordinate : `~astropy.coordinates.SkyCoord`, `~sunpy.coordinates.frames.Helioprojective`
+        The input coordinate.  The frame property of the input
+        coordinate must be helioprojective.
+
+    scale : `~astropy.units.Quantity`
+        The pixel scale size in units of arcseconds.
+
+    Returns
+    -------
+    is_on_disk : `~bool`
+        Returns True if the coordinate is on disk, False otherwise.
+    """
+    # Calculate the radii of every pixel in helioprojective Cartesian
+    # co-ordinate distance units.
+    if not isinstance(coordinate.frame, Helioprojective):
+        raise ValueError("Input coordinate must be in the Helioprojective Cartesian frame.")
+    return u.R_sun * (np.sqrt(coordinate.Tx ** 2 + coordinate.Ty ** 2) / scale) < 1 * u.R_sun
+
+
+def on_disk_bounding_coordinates(smap):
+    """
+    Returns the the bottom left and top-right coordinates of a minimal
+    rectangular region that contains all the on-disk pixels in the input map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
+
+    Returns
+    -------
+    (bl, tr) : `~list`
+        Returns the bottom left and top right coordinates that bound the
+        spatial location of all the on-disk pixels.
+    """
+    # Check that the input map is not all off disk.
+    if is_all_off_disk(smap):
+        raise ValueError("The entire map is off disk.")
+
+    # Get all the coordinates from the input map
+    coordinates = all_coordinates_from_map(smap)
+
+    # Find which coordinates are on the disk
+    on_disk = coordinate_is_on_disk(coordinates, smap.rsun_obs)
+    on_disk_coordinates = coordinates[on_disk]
+
+    # The bottom left and top right coordinates that contain
+    # the on disk coordinates.
+    bl = SkyCoord(np.nanmin(on_disk_coordinates.Tx), np.nanmin(on_disk_coordinates.Ty),
```

Comment:
You could combine these into one SkyCoord of length two here as well?

Reply:
done